tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Wcf/TaskMethodInvokerIntegration.cs

```diff
@@ -38,28 +40,12 @@ public class TaskMethodInvokerIntegration
         /// <returns>Calltarget state value</returns>
         internal static CallTargetState OnMethodBegin<TTarget>(TTarget instance, object instanceArg, object[] inputs)
         {
-            // TODO Just use the OperationContext.Current object to get the span information
-            // context.IncomingMessageHeaders contains:
-            //  - Action
-            //  - To
-            //
-            // context.IncomingMessageProperties contains:
-            // - ["httpRequest"] key to find distributed tracing headers
             if (!Tracer.Instance.Settings.IsIntegrationEnabled(WcfCommon.IntegrationId) || !Tracer.Instance.Settings.DelayWcfInstrumentationEnabled || WcfCommon.GetCurrentOperationContext is null)
             {
                 return CallTargetState.GetDefault();
             }
 
-            var operationContext = WcfCommon.GetCurrentOperationContext?.Invoke();
-            if (operationContext != null && operationContext.TryDuckCast<IOperationContextStruct>(out var operationContextProxy))
-            {
-                var useWcfWebHttpResourceNames = Tracer.Instance.Settings.WcfWebHttpResourceNamesEnabled;
-                return new CallTargetState(WcfCommon.CreateScope(operationContextProxy.RequestContext, useWcfWebHttpResourceNames));
-            }
-            else
-            {
-                return CallTargetState.GetDefault();
-            }
+            return new CallTargetState(Tracer.Instance.ActiveScope as Scope);
```

Comment:
Note: The new AfterReceiveRequestIntegration and BeforeSendReplyIntegration integrations start/stop the span. This will only decorate the existing span with exception information.